app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
I will have a look to convince myself about the order of magnitude those things will run in so let's hold off on that for a bit ( As I say I am being nervous and I will try and see if I can back it up with data ).